src/gpu/nvidia/cudnn_inner_product.cpp

```diff
@@ -36,16 +37,12 @@ status_t cudnn_inner_product_fwd_t::execute(const exec_ctx_t &ctx) const {
     return cuda_stream->interop_task([&](::sycl::handler &cgh) {
         using scratch_acc_t = ::sycl::accessor<uint8_t, 1,
                 ::sycl::access::mode::read_write, sycl::compat::target_device>;
-        using read_acc_t = ::sycl::accessor<uint8_t, 1,
-                ::sycl::access::mode::read, sycl::compat::target_device>;
-        auto src_acc = CTX_IN_ACCESSOR(DNNL_ARG_SRC);
-        auto wei_acc = CTX_IN_ACCESSOR(DNNL_ARG_WEIGHTS);
-        std::shared_ptr<read_acc_t> bias_acc;
-        if (pd()->with_bias()) {
-            bias_acc = std::make_shared<read_acc_t>(
-                    CTX_IN_ACCESSOR(DNNL_ARG_BIAS));
-        }
-        auto dst_acc = CTX_OUT_ACCESSOR(DNNL_ARG_DST);
+        auto arg_src = CTX_IN_SYCL_MEMORY(DNNL_ARG_SRC);
+        auto arg_wei = CTX_IN_SYCL_MEMORY(DNNL_ARG_WEIGHTS);
+
+        sycl_memory_arg<::sycl::access::mode::read> arg_bias;
+        if (pd()->with_bias()) { arg_bias = CTX_IN_SYCL_MEMORY(DNNL_ARG_BIAS); }
```

Comment:
We can do `auto arg_bias = CTX_IN_SYCL_MEMORY(DNNL_ARG_BIAS)` unconditionally, `arg_bias` will be empty.

Reply:
done